test/features/prebooking.js

```diff
@@ -0,0 +1,303 @@
+      var expectedMaleCentre = 'bigone';
+
+      before(function () {
+        global.testConfig.initializeBarrelsFixtures = false;
+        return global.initializeBarrelsFixtures();
+      });
+
+      after(function () {
+        global.testConfig.initializeBarrelsFixtures = true;
+      });
+
+      Given(`a prebooking with cid id "${payload.Output[0].cid_id}" has already occurred`, () =>
+        createRequest(payload, '/depmu_entry/prebooking').then(
+          () => findPrebookingByCID(payload.Output[0].cid_id).then((models) => {
+            expect(models.length).to.equal(1);
+            expect(models[0].cid_id).to.equal(parseInt(payload.Output[0].cid_id));
+          })
+        )
+      );
+
+      When(`new valid prebookings occur`, () => createRequest(followingPayload, '/depmu_entry/prebooking'));
+
+      Then('previous prebookings recorded are removed',
+        () => findPrebookingByCID(payload.Output[0].cid_id).then((models) => expect(models.length).to.equal(0))
+      );
+
+      And(`new female prebookings are created`, () =>
+        Prebooking.find({where: {gender: 'female'}}).then((models) => expect(models.length).to.equal(2))
+      );
+      And(`new male prebookings are created`, () =>
+        Prebooking.find({where: {gender: 'male'}}).then((models) => expect(models.length).to.equal(2))
+      );
+
+    });
+
+    Scenario('New Invalid Pre-bookings are ignored', () => {
+      var followingPayload = {
+        Output: [{
+          timestamp: validTimestamp,
+          location: 'smallone female holding',
+          cid_id: '111'
+        }]
+      };
+
+      before(function () {
+        global.testConfig.initializeBarrelsFixtures = false;
+        return global.initializeBarrelsFixtures();
+      });
+
+      after(function () {
+        global.testConfig.initializeBarrelsFixtures = true;
+      });
+
+      Given(`a prebooking with cid id "${payload.Output[0].cid_id}" has already occurred`, () =>
+        createRequest(payload, '/depmu_entry/prebooking').then(
+          () => findPrebookingByCID(payload.Output[0].cid_id).then((models) => {
+            expect(models.length).to.equal(1);
+            expect(models[0].cid_id).to.equal(parseInt(payload.Output[0].cid_id));
+          })
+        )
+      );
+
+      When(`a new invalid prebooking occurs`, () => createRequest(followingPayload, '/depmu_entry/prebooking', 400));
+
+      Then('previous prebookings recorded are retained',
+        () => findPrebookingByCID(payload.Output[0].cid_id).then((models) => expect(models.length).to.equal(1))
+      );
+
+      And(`new invalid prebooking is not created`,
+        () => findPrebookingByCID(followingPayload.cid_id).then((models) => expect(models.length).to.equal(0))
+      );
+
+    });
+
+    Scenario('Remove Existing Pre-booking when Movement In Order occurs', () => {
+      var validTimestamp = moment().set({hour: 7, minute: 0, second: 0}).format("DD/MM/YYYY H:mm:ss");
```

Comment:
whats wrong with the validTimestamp you made at the start? is there a need to have a new one in this scope?
its probably fine, just reads a bit odd to reuse the same varname